lib/Project.js

```diff
@@ -708,6 +708,9 @@ Project.prototype.close = function(cb) {
             logger.trace("Collecting extracted strings from " + this.fileTypes[i].name());
             extracted.addAll(this.fileTypes[i].getExtracted().getBy({
                 sourceLocale: this.sourceLocale
+            }).filter(function(res) {
+                // no source means nothing to translate, so don't need those resources
+                return res.source || res.sourceArray || res.sourceStrings;
```

Comment:
Empty array and empty object are truthy so they would still pass this check - is this intentional?

Reply:
The root problem is the translation file for salesforce metaxml sometimes only contains the keys of resources to translate and no source string, so we create a resource with only a key and we have to go find the source strings in other xml files. (All resources in the Salesforce plugin are strings.) If, at the end of searching those other xml files, we still have a resource with a key but no source string, then that's one we filter out.

The xliff code will not emit any trans-units for empty array or plural entries, so that part is okay. The array and plural checks are just there for safety. We mostly care about the missing or empty string resources.